influxdb3_catalog/src/serialize.rs

```diff
@@ -70,6 +70,9 @@ impl From<DatabaseSnapshot> for DatabaseSchema {
             name: snap.name,
             tables,
             table_map,
+            // todo: check if it's right to default to false here,
+            //       not sure where this is called
+            deleted: false,
```

Comment:
Should be fine to add a `deleted` field on the `DatabaseSnapshot` struct, then base it off of that.

Reply:
This bit wasn't too obvious when I saw the `From` impls, I see all the types have `..Snapshot` structs for serializing and deserializing, is that avoiding to build deserializer by hand? I guess in `Deserialize` impls for partial structs (`..Snapshot`) you can use `deserialize` directly and then just build the relevant fields for the target type manually without involving visitor impl - is that the motivation?

Reply:
Yeah, that about sums it up. The main motivation is the bi-directional maps that map from ID to name and vice versa. Those only need to be held in memory because the information in that bi-directional map is contained in the main map that maps the ID to the whole object, and which is serialized with the `SerdeVecMap`. No need to also serialize the bi-directional map and waste bytes.

Earlier on in the project, the motivation for the `_Snapshot` types was to capture the info that is contained in the arrow `Schema`, since we could not rely on its `Serialize`/`Deserialize` impls being stable.